url56b1dc76d3_wufunxiang_Libs_tgz-pJ8-org_msgpack_util_TemplatePrecompilerJ8/src/org/msgpack/MessagePack.java

```diff
@@ -0,0 +1,837 @@
+        return registry.unregister(type);
+    }
+
+    /**
+     * Unregisters all {@link org.msgpack.template.Template} objects that have
+     * been registered in advance.
+     * 
+     * @since 0.6.0
+     */
+    public void unregister() {
+        registry.unregister();
+    }
+
+    /**
+     * Looks up a {@link org.msgpack.template.Template} object, which is
+     * serializer/deserializer associated by specified class.
+     * 
+     * @since 0.6.0
+     * @param type
+     * @return
+     */
+    @SuppressWarnings("unchecked")
+    public <T> Template<T> lookup(Class<T> type) {
+        return registry.lookup(type);
+    }
+
+    public Template<?> lookup(Type type) {
+        return registry.lookup(type);
+    }
+
+    private static final MessagePack globalMessagePack = new MessagePack();
+
+    /**
+     * Serializes specified object and returns the byte array.
+     * 
+     * @deprecated {@link MessagePack#write(Object)}
+     * @param v
+     * @return
+     * @throws IOException
+     */
+    @Deprecated
+    public static byte[] pack(Object v) throws IOException {
+        return globalMessagePack.write(v);
+    }
+
+    /**
+     * Serializes specified object to output stream.
+     * 
+     * @deprecated {@link MessagePack#write(OutputStream, Object)}
+     * @param out
+     * @param v
+     * @throws IOException
+     */
+    @Deprecated
+    public static void pack(OutputStream out, Object v) throws IOException {
+        globalMessagePack.write(out, v);
+    }
+
+    /**
+     * Serializes object by specified template and return the byte array.
+     * 
+     * @deprecated {@link MessagePack#write(Object, Template)}
+     * @param v
+     * @param template
+     * @return
+     * @throws IOException
+     */
+    @Deprecated
+    public static <T> byte[] pack(T v, Template<T> template) throws IOException {
+        return globalMessagePack.write(v, template);
+    }
+
+    /**
+     * Serializes object to output stream. The object is serialized by specified
+     * template.
+     * 
+     * @deprecated {@link MessagePack#write(OutputStream, Object, Template)}
+     * @param out
+     * @param v
+     * @param template
+     * @throws IOException
+     */
+    @Deprecated
+    public static <T> void pack(OutputStream out, T v, Template<T> template)
+            throws IOException {
+        globalMessagePack.write(out, v, template);
+    }
+
+    /**
+     * Converts byte array to {@link org.msgpack.type.Value} object.
+     * 
+     * @deprecated {@link MessagePack#read(byte[])}
+     * @param bytes
+     * @return
+     * @throws IOException
+     */
+    @Deprecated
+    public static Value unpack(byte[] bytes) throws IOException {
+        return globalMessagePack.read(bytes);
+    }
+
+    @Deprecated
+    public static <T> T unpack(byte[] bytes, Template<T> template) throws IOException {
+        BufferUnpacker u = new MessagePackBufferUnpacker(globalMessagePack).wrap(bytes);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
Analysis of this package determined that this line of code might contain a resource leak. A resource leak can slow down or crash your system.

**Fix**
Consider closing the following resource in a try-finally block: *u*. Currently, there are execution paths that do not contain closure statements, for example, when Template.read throws an exception.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).